src/components/DataViewer/DataViewer.tsx

```diff
@@ -1,78 +1,58 @@
-import CloseIcon from "@mui/icons-material/Close";
+"use client";
+
 import React, { ReactElement } from "react";
-import Modal from "react-modal";
 import styled from "styled-components";
+import Modal from "@/components/Modal/Modal";
 
 export interface Data {
-    [key: string]: string | number | null;
+    [key: string]: string[] | string | number | boolean | null;
 }
 
-const StyledModal = styled(Modal)`
-    display: flex;
-    flex-direction: column;
-    position: absolute;
-    top: 50%;
-    left: 50%;
-    transform: translate(-50%, -50%);
-    border: 0;
-    border-radius: 5px;
-    outline: none;
-    padding: 0;
-    width: 80%;
-    max-height: 80%;
-    background: rgb(255, 255, 255);
-    box-shadow: rgb(200, 200, 200) 0 2px 10px;
-    overflow: hidden;
-`;
-const Header = styled.h3`
-    display: flex;
-    justify-content: space-between;
-    align-items: center;
-    background-color: #eeeeee;
-    padding: 0.5rem;
-    margin-top: 0;
-`;
-const Content = styled.div`
-    overflow: scroll;
-    margin: 1.5rem;
-    margin-top: 1rem;
-`;
 const Key = styled.div`
-    color: grey;
-    font-size: small;
+    // TODO VFB-16 Change to using theme with palettes of colours (dark font matching with this background)
+    color: ${(props) => props.theme.foregroundColor};
+    // TODO VFB-16 Change to using theme with palettes of colours
+    background-color: ${(props) => props.theme.secondaryBackgroundColor}40;
+    display: inline-block;
+    border-radius: 0.7em;
+    padding: 0.2em 0.5em;
 `;
+
 const Value = styled.div`
-    font-size: medium;
+    padding: 0.2em 0.5em;
 `;
+
 const EachItem = styled.div`
-    padding-bottom: 1rem;
+    padding-bottom: 1em;
 `;
-const CloseButton = styled.button`
-    border: 0;
-    background-color: red;
-    color: white;
-    display: flex;
-    justify-content: center;
+
+const ContentDiv = styled.div`
+    width: 1000px;
+    max-width: 100%;
 `;
 
-interface ClearButtonProps {
-    closeModal: () => void;
-}
+const valueIsEmpty = (value: string[] | string | number | boolean | null): boolean => {
+    return (Array.isArray(value) && value.length === 0) || value === "" || value === null;
+};
 
-const ClearButton: React.FC<ClearButtonProps> = (props) => {
-    return (
-        <CloseButton onClick={props.closeModal}>
-            <CloseIcon />
-        </CloseButton>
-    );
+const formatDisplayValue = (value: any): string => {
+    if (valueIsEmpty(value)) {
+        return "-";
+    } else if (typeof value === "boolean") {
```

Comment:
`else` is not needed if there's a `return` - the code should just be like
```tsx
if (valueIsEmpty(value)) {
   return "-";
}
if (/* ... */) {
   return;
}
return;
```

Reply:
Yeah, me and Chloe agreed on this point, but thought that you would prefer the explicit `else` cases

Reply:
cool - I think `else` is a relatively 'unclean' practice! It's a sign that the if could be extracted to functions with `return`s instead :)